src/vivarium_csu_ltbi/components/observers.py

```diff
@@ -113,7 +113,7 @@ def on_collect_metrics(self, event):
 
             for transition in ltbi_globals.HIV_TB_TRANSITIONS:
                 from_state, to_state = transition.split('_to_')
-                event_this_step = ((pop_in_category[f'{to_state}_event_time'] == self.clock())
+                event_this_step = ((pop_in_category[f'{to_state}_event_time'] == event.time)
```

Comment:
It was impossible to be in this filter before.  During time step prepare we set the previous state column as the current state column (which makes sense, that's the right time to do it).  Then we update the state table with their new state and the event time, which is the current time plus the size of a time step as well as their new state. By the time we'd want to check if they've transitioned (on the next step) we've already reset their previous state to the current state.

Technically, we're counting the events a time step before they can occur since the clock hasn't actually advanced at this point.  That's a more difficult problem to fix and the error it introduces is on the scale of 1/num_time_steps which here is less than 0.1%